tests/console/validate_fs_table.pm

```diff
@@ -0,0 +1,65 @@
+
+use strict;
+use warnings;
+use base "opensusebasetest";
+use testapi;
+use scheduler 'get_test_suite_data';
+use Test::Assert ':all';
+
+sub run {
+
+    select_console "root-console";
+    my $test_data = get_test_suite_data();
+    if (defined $test_data->{table_type}) {
+        # Validate that the partition table type is the expected one.
+        my $table_type = script_output("parted -l | sed -n -e 4p | awk {'print \$3'}");
+        assert_equals($test_data->{table_type}, $table_type, "Partition table type does not correspont to the expected one.");
+    }
+    foreach my $disk (@{$test_data->{disks}}) {
+        foreach my $partition (@{$disk->{partitions}}) {
+            if ($_->{role} eq 'swap') {
+                record_info("Check $_->{role}", "Verify the '$_->{role}' partition was created and the filesystem is '$partition->{formatting_options}->{filesystem}'");
+                # Validate that the partition is created.
+                assert_script_run("lsblk -n | grep SWAP", fail_message => "Partition SWAP was not created.");
+                # Validate that the filesystem type is the expected one.
+                my $fstype = script_output("cat /etc/fstab | grep swap | awk '{print \$3}'");
+                assert_equals($partition->{formatting_options}->{filesystem}, $fstype, "File system on '$_->{role}' partition does not correspond to the expected one.");
+            } elsif (($_->{role} eq 'data') || ($partition->{mounting_options}->{mount_point} eq '/')) {
+                record_info("Check $partition->{mounting_options}->{mount_point}", "Verify the '$partition->{mounting_options}->{mount_point}' partition was created and the filesystem is '$partition->{formatting_options}->{filesystem}'");
+                # Validate that the partition is created.
+                assert_script_run("lsblk -n | grep $partition->{mounting_options}->{mount_point}\$", fail_message => "Partition $partition->{mounting_options}->{mount_point} was not created.");
+                # Validate that the filesystem type is the expected one.
+                my $fstype = script_output("df -PT $partition->{mounting_options}->{mount_point} | grep -v \"Filesystem\" | awk '{print \$2}'");
+                assert_equals($partition->{formatting_options}->{filesystem}, $fstype, "File system on '$partition->{mounting_options}->{mount_point}' partition does not correspond to the expected one.");
+                # Test the ability to read and write files on the partition.
+                assert_script_run("echo Hello > $partition->{mounting_options}->{mount_point}/emptyfile", fail_message => 'Failure while writing in ' . $partition->{mounting_options}->{mount_point});
+                assert_script_run("cat $partition->{mounting_options}->{mount_point}/emptyfile", fail_message => 'Failure while reading from ' . $partition->{mounting_options}->{mount_point});
+            } elsif (sprintf("%d", $_->{size}) le 300) {
```

Comment:
`sprintf` returns a string formatted by the usual printf conventions of the C library function sprint and `le` operator is for string comparison. So you are trying to check that '300mb' is less or equal than '300', which should be the opposite, '300mb' is alphabetically greater than '300', but for me if fails to use sprintf with that value, but it looks like you wanted numeric comparison. Could you clarify that? thx

Reply:
Yes, I wanted to convert it to numeric, because I was under the impression that comparing strings might end up with warning, but maybe I am confusing 'le' with '<='. 
I am not sure if it is needed to check if /boot/efi partition is created, as a successful first boot means that indeed, it was created. What do you think?

Reply:
right, we can assume is fine and it is already tested. It could be tricky because you would need to store the numeric value in some data key and the unit in another or doing some regex. Perhaps better to skip it.